confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -186,32 +308,30 @@ def get_schema(self, schema_id):
             SchemaRegistryClientError: If schema can't be found
 
         .. _Schema Registry API Reference:
-            https://docs.confluent.io/current/schema-registry/develop/api.html#get--schemas-ids-int-%20id
+            https://docs.confluent.io/current/schema-registry/develop/api.html#get--'schemas/ids/{}?format=serialized'-ids-int-%20id
 
         """  # noqa: E501
-        version = self.id_to_schema.get(schema_id, None)
+        schema = self.cache.get_schema(schema_id)
+        if schema is not None:
+            return schema
 
-        if version is not None:
-            return version
+        response = self._rest_client.GET('schemas/ids/{}'.format(schema_id))['schema']
+        schema = Schema.from_json(response)
```

Comment:
For readability I'd put the schema deref here: `..from_json(response['schema'])`